src/analysis/processing/qgsalgorithmexporttospreadsheet.cpp

```diff
@@ -0,0 +1,262 @@
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmexporttospreadsheet.h"
+#include "qgsgeometryengine.h"
+#include "qgsogrutils.h"
+#include "qgsvectorfilewriter.h"
+#include "qgsvectorlayer.h"
+#include "qgssettings.h"
+#include "qgsapplication.h"
+#include "qgsfieldformatterregistry.h"
+#include "qgsfieldformatter.h"
+
+///@cond PRIVATE
+
+class FieldValueConverter : public QgsVectorFileWriter::FieldValueConverter
+{
+  public:
+    FieldValueConverter( QgsVectorLayer *vl )
+      : mLayer( vl )
+    {
+      QStringList formattersAllowList{ QStringLiteral( "KeyValue" ),
+                                       QStringLiteral( "List" ),
+                                       QStringLiteral( "ValueRelation" ),
+                                       QStringLiteral( "ValueMap" ) };
+
+      for ( int i = 0; i < mLayer->fields().count(); ++i )
+      {
+        const QgsEditorWidgetSetup setup = mLayer->fields().at( i ).editorWidgetSetup();
+        const QgsFieldFormatter *fieldFormatter = QgsApplication::fieldFormatterRegistry()->fieldFormatter( setup.type() );
+        if ( formattersAllowList.contains( fieldFormatter->id() ) )
+          mIndicesToConvert << i;
+      }
+    }
+
+    QgsField fieldDefinition( const QgsField &field ) override
+    {
+      if ( !mLayer )
+        return field;
+
+      int idx = mLayer->fields().indexFromName( field.name() );
+      if ( mIndicesToConvert.contains( idx ) )
+      {
+        return QgsField( field.name(), QVariant::String );
+      }
+      return field;
+    }
+
+    QVariant convert( int i, const QVariant &value ) override
+    {
+      if ( !mIndicesToConvert.contains( i ) )
+        return value;
+
+      const QgsEditorWidgetSetup setup = mLayer->fields().at( i ).editorWidgetSetup();
+      const QgsFieldFormatter *fieldFormatter = QgsApplication::fieldFormatterRegistry()->fieldFormatter( setup.type() );
+      return fieldFormatter->representValue( mLayer.data(), i, setup.config(), QVariant(), value );
+    }
+
+    FieldValueConverter *clone() const override
+    {
+      return new FieldValueConverter( *this );
+    }
+
+  private:
+    QPointer< QgsVectorLayer > mLayer;
+    QList< int > mIndicesToConvert;
+};
+
+QString QgsExportToSpreadsheetAlgorithm::name() const
+{
+  return QStringLiteral( "exporttospreadsheet" );
+}
+
+QString QgsExportToSpreadsheetAlgorithm::displayName() const
+{
+  return QObject::tr( "Export to spreadsheet" );
+}
+
+QStringList QgsExportToSpreadsheetAlgorithm::tags() const
+{
+  return QObject::tr( "microsoft,excel,xls,xlsx,calc,open,office,libre,ods" ).split( ',' );
+}
+
+QString QgsExportToSpreadsheetAlgorithm::group() const
+{
+  return QObject::tr( "Database" );
```

Comment:
It's a hard alg to classify in our pre-existing group, but I'm not sure database is the right option. Maybe layer tools would be better.